docs/plugin/custom-executor.md

```diff
@@ -0,0 +1,212 @@
+
+### Develop plugin business logic
+
+1. Create an executor plugin directory:
+
+   ```bash
+   mkdir botkube-plugins && cd botkube-plugins
+   ```
+
+2. Initialize the Go module:
+
+   ```bash
+   go mod init botkube-plugins
+   ```
+
+3. Create the `main.go` file for the `echo` executor with the following template:
+
+   ```go
+   cat << EOF > main.go
+   package main
+
+   import (
+     "context"
+     "fmt"
+
+     "github.com/hashicorp/go-plugin"
+     "github.com/kubeshop/botkube/pkg/api"
+     "github.com/kubeshop/botkube/pkg/api/executor"
+   )
+
+   // EchoExecutor implements the Botkube executor plugin interface.
+   type EchoExecutor struct{}
+
+   func main() {
+     executor.Serve(map[string]plugin.Plugin{
+       "echo": &executor.Plugin{
+         Executor: &EchoExecutor{},
+       },
+     })
+   }
+   EOF
+   ```
+
+   This template code imports required packages and registers `EchoExecutor` as the gRPC plugin. At this stage, the `EchoExecutor` service doesn't implement the required [Protocol Buffers](https://github.com/kubeshop/botkube/blob/main/proto/executor.proto) contract. We will add the required methods in the next steps.
+
+4. Download imported dependencies:
+
+   ```bash
+    go mod tidy
+   ```
+
+5. Add the required `Metadata` method:
+
+   ```go
+   // Metadata returns details about the Echo plugin.
+   func (EchoExecutor) Metadata(context.Context) (api.MetadataOutput, error) {
+     return api.MetadataOutput{
+       Version:     "0.0.1",
+       Description: "Echo sends back the command that was specified.",
+     }, nil
+   }
+   ```
+
+   The `Metadata` method returns basic information about your plugin. This data is used when the plugin index is generated in an automated way. You will learn more about that in the next steps.
+
+6. Add the required `Execute` method:
+
+   ```go
+   // Execute returns a given command as a response.
+   func (EchoExecutor) Execute(_ context.Context, in executor.ExecuteInput) (executor.ExecuteOutput, error) {
+    return executor.ExecuteOutput{
+      Data: fmt.Sprintf("Echo: %s", in.Command),
+    }, nil
+   }
+   ```
+
+   The `Execute` method is the heart of your executor plugin. This method runs your business logic and returns the output as plain text. Next, the Botkube core sends back the response to a given communication platform.
+
+   For each `Execute` method call, Botkube attaches the list of associated configurations. You will learn more about that in the [**Passing configuration to your plugin**](#passing-configuration-to-your-plugin) section.
+
+## Build plugin binaries
+
+Now it's time to build your plugin. For that purpose we will use GoReleaser. It simplifies building Go binaries for different architectures.
+
+:::note
+Instead of GoReleaser, you can use another tool of your choice. The important thing is to produce the binaries for the architecture of the host platform where Botkube is running.
+:::
+
+1. Create the GoReleaser configuration file:
```

Comment:
Should we mentioned about possible platform dependency? Think about I am windows user, and without QEMU, I will get an error with build, I assume :)

Reply:
I mentioned that in the "info" above: "The important thing is to produce the binaries for the architecture of the host platform where Botkube is running."

But maybe I will change it and move it directly to this step, so this relation will be easier to find